toolsrc/src/vcpkg-test/format-manifest.cpp

```diff
@@ -0,0 +1,127 @@
+#include <catch2/catch.hpp>
+
+#include <vcpkg/base/files.h>
+#include <vcpkg/base/stringview.h>
+
+#include <vcpkg/commands.format-manifest.h>
+
+#include <algorithm>
+#include <system_error>
+
+using namespace vcpkg;
+using namespace vcpkg::Commands::FormatManifest;
+
+namespace
+{
+    struct ListOfExistingFiles : Files::ITestFileExists
+    {
+        std::vector<fs::path> allowed_paths;
+
+        ListOfExistingFiles(std::vector<fs::path>&& allowed_paths_) : allowed_paths(std::move(allowed_paths_)) { }
+
+        virtual bool exists(const fs::path& test, std::error_code&) const override
+        {
+            return std::any_of(allowed_paths.begin(), allowed_paths.end(), [&](const fs::path& candidate) {
+                return test == candidate;
+            });
+        }
+    };
+
+#if defined(_WIN32)
+    StringView existing_absolute{"C:\\hello"};
+    StringView missing_absolute{"C:\\hello\\world"};
+#define SEPARATOR "\\"
+#else // ^^^ defined(_WIN32) // !defined(_WIN32) vvv
+    StringView existing_absolute{"/hello"};
+    StringView missing_absolute{"/hello/world"};
+#define SEPARATOR "/"
+#endif // ^^^ !defined(_WIN32)
+
+    const auto existing_absolute_path = fs::u8path(existing_absolute);
+    const auto missing_absolute_path = fs::u8path(missing_absolute);
+
+    const auto original_cwd = existing_absolute_path / fs::u8path("cwd");
+    const auto ports = existing_absolute_path / fs::u8path("ports");
```

Comment:
I do not like this; it's super unnecessarily complex. I'd do something like:

```suggestion
#if defined(_WIN32)
    auto fs_root = fs::u8path(R"(C:\)");
#else
    auto fs_root = fs::u8path("/");
#endif

    auto existing_absolute = fs_root / fs::u8path("hello");
    auto missing_absolute = fs_root / fs::u8path("hello/world");

    const auto original_cwd = existing_absolute_path / fs::u8path("cwd");
    const auto ports = existing_absolute_path / fs::u8path("ports");
```

Reply:
That makes existing_absolute and missing_absolute have the wrong type.

Reply:
You don't ever use these again, so it's not important.

Reply:
Yes I do, on 72 and 79

Reply:
oh, these are global.

Reply:
I'm not sure what you mean by "still remove the rest" -- everything here is used. The only other obvious change here is your removal of SEPARATOR but that causes the string comparisons in `ListOfExistingFiles::exists` to fail since the paths the tool generates always have the platform's preferred separator rather than `/`. I could do something like changing:

    "ambiguous-port" SEPARATOR "vcpkg.json"

into:

    fs::u8path("ambiguous-port") / fs::u8path("vcpkg.json")

but I don't believe that is an improvement.